inc/autoload/class-add-admin-favicon.php

```diff
@@ -78,17 +78,21 @@ public function __construct() {
 		 *
 		 * @type array
 		 */
-		$hooks = (array) apply_filters( 'multisite_enhancements_favicon', self::$favicon_hooks );
+		if ( '1' === Multisite_Enhancements_Settings::get_settings( 'add-favicon' ) ) {
```

Comment:
We should remove this logic, check for settings in a separate method, so that we get an empty constructor without logic. Thus, the goal should have the principle get simpler; I mean with should not wrap in `if (settings === 1) {}` we should more check for settings and if not, return from the method. An alternate is a solid method inside the settings-class that get always the boolean from a setting, so that we ask inside this settings-class for: is the feature active.
Do you know what I mean. See this for an example https://github.com/bueltge/adminimize/blob/refactor/src/Settings/SettingsRepository.php#L23

Reply:
Refactored this in https://github.com/bueltge/wordpress-multisite-enhancements/pull/63/commits/25674923aaff58b6405a88826173af16d2f81c90 - what do you think?